addons/web_editor/static/src/js/editor/odoo-editor/test/spec/utils.test.js

```diff
@@ -1541,8 +1542,143 @@ describe('Utils', () => {
+            const [p] = insertTestHtml(
+                `<p>abc</p>`,
+            );
+            const editable = p.parentElement;
+            const abc = p.firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([abc, 0]);
+            [node, offset] = getDeepestPosition(editable, 1);
+            window.chai
+                .expect([node, offset])
+                .to.eql([abc, 3]);
+        });
+        it('should get deepest position within nested formatting tags', () => {
+            const [p] = insertTestHtml(
+                `<p><span><b><i><u>abc</u></i></b></span></p>`,
+            );
+            const editable = p.parentElement;
+            const abc = p.firstChild.firstChild.firstChild.firstChild.firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([abc, 0]);
+            [node, offset] = getDeepestPosition(editable, 1);
+            window.chai
+                .expect([node, offset])
+                .to.eql([abc, 3]);
+        });
+        it('should get deepest position in multiple paragraph', () => {
+            const [p1, p2] = insertTestHtml(
+                `<p>abc</p><p>def</p>`,
+            );
+            const editable = p1.parentElement;
+            const abc = p1.firstChild;
+            const def = p2.firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([abc, 0]);
+            [node, offset] = getDeepestPosition(editable, 1);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 0]);
+            [node, offset] = getDeepestPosition(editable, 2);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 3]);
+        });
+        it('should get deepest position for node with invisible element', () => {
+            const [p1] = insertTestHtml(
+                `<p></p><p>def</p>`,
+            );
+            const editable = p1.parentElement;
+            const def = editable.lastChild.firstChild;
+            let [node, offset] = getDeepestPosition(editable, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 0]);
+            [node, offset] = getDeepestPosition(editable, 2);
+            window.chai
+                .expect([node, offset])
+                .to.eql([def, 3]);
+        });
+        it('should get deepest position for invisible block element', () => {
+            const [p] = insertTestHtml(
+                `<p></p><span>def</span>`,
+            );
+            const [node, offset] = getDeepestPosition(p, 0);
+            window.chai
+                .expect([node, offset])
+                .to.eql([p, 0]);
```

Comment:
No noting related to span here that changes the result it is the fact that you target the `P` tag directly.

> What would happen if we targeted P, 0 in the previous test with `<p></p><p>def</p>`?

It would be the same result P, 0 but if you targeted editor, 0 in the same test it should be text node `def`.
![image](https://github.com/odoo/odoo/assets/120459801/70a3bc86-a127-4c19-a60c-bd51af54aabb)